src/Mod/Sketcher/Gui/ViewProviderSketch.cpp

```diff
@@ -3899,4 +3816,156 @@ bool ViewProviderSketch::isInEditMode() const
 {
     return editCoinManager != nullptr;
 }
+void ViewProviderSketch::generateContextMenu() {
+    int selectedEdges = 0;
+    int selectedLines = 0;
+    int selectedConics = 0;
+    int selectedPoints = 0;
+    int selectedConstraints = 0;
+
+    Gui::MenuItem menu;
+    menu.setCommand("Sketcher context");
+
+    std::vector<Gui::SelectionObject> selection = Gui::Selection().getSelectionEx(0, Sketcher::SketchObject::getClassTypeId());
+
+    // if something is selected, count different elements in the current selection
+    if (selection.size() > 0) {
+        const std::vector<std::string> SubNames = selection[0].getSubNames();
+
+        for (auto name : SubNames) {
```

Comment:
There is a Lint warning here, see below my message.

> Missing reference in range-for with non trivial type (std::basic_string<char>) [-Wclazy-range-loop-reference]

I think you need to change this to : `for (auto& name : SubNames) {` ie add a `&`